src/services/validatorSchema.ts

```diff
@@ -0,0 +1,160 @@
+import * as yup from "yup";
+
+export const UserSchema = yup.object({
+    user_id: yup
+        .string()
+        .strict()
+        .typeError("user_id must be a string.")
+        .defined("The user_id (string) must be provided.")
+});
+
+export const ItemSchema = yup.object({
+    item_id: yup
+        .number()
+        .typeError("item_id must be an integer.")
+        .positive("item_id must be a positive integer.")
+        .defined("The item_id (positive integer) must be provided.")
+});
+
+export const InventorySchema = yup.object({
+    inventory_item_id: yup
+        .string()
+        .strict()
+        .typeError("inventory_item_id must be a string.")
+        .defined("The inventory_item_id (string) must be provided.")
+});
+
+export const DataSchema = yup.object({
+    data: yup.lazy((value) => {
+        if (value === undefined || value === null)
+            return yup
+                .object()
+                .required("The data (object of string(s)) must be provided.");
+        else {
+            const schema = Object.keys(value).reduce(
+                (acc: any, curr: string) => {
+                    acc[curr] = yup
+                        .string()
+                        .strict()
+                        .typeError("data's field must be a string.")
+                        .required(
+                            "The data's field (string) must be provided."
+                        );
+                    return acc;
+                },
+                {}
+            );
+            return yup
+                .object()
+                .shape(schema)
+                .required("The data (object of string(s)) must be provided.");
+        }
+    })
+});
+
+export const JsonUserSchema = yup.object({
+    user_id: yup
+        .number()
+        .typeError("user_id must be an integer.")
+        .positive("user_id must be a positive integer.")
+        .defined("The user_id (positive integer) must be provided."),
+    inventory_address: yup
```

Comment:
when I create a new User, at first I don't have an inventory address, only a user id. The inventory address is added to the new user when the user is created.
So we can't require inventory_address when creating the user. I think we should distinguish between the "user schema" and the "create user request payload"? I don't mind having a yup validation for request payloads directly put at the top of the relevant service,